ui-packages/packages/management-console/src/components/Molecules/ProcessListToolbar/ProcessListToolbar.tsx

```diff
@@ -16,70 +16,191 @@ import {
   DropdownToggle,
   DropdownToggleCheckbox,
   DropdownItem,
-  DropdownPosition
+  DropdownPosition,
+  OverflowMenuControl,
+  OverflowMenuContent,
+  KebabToggle,
+  OverflowMenu,
+  OverflowMenuItem
 } from '@patternfly/react-core';
 import { FilterIcon, SyncIcon } from '@patternfly/react-icons';
 import _ from 'lodash';
 import './ProcessListToolbar.css';
 import { GraphQL } from '@kogito-apps/common';
 import ProcessInstanceState = GraphQL.ProcessInstanceState;
-import { handleAbortAll } from '../../../utils/Utils';
-
 type filterType = {
   status: ProcessInstanceState[] | string[];
   businessKey: string[];
 };
+
+export interface ProcessInstanceBulkList {
+  [key: string]: GraphQL.ProcessInstance;
+}
+
+import ProcessListModal from '../../Atoms/ProcessListModal/ProcessListModal';
+import { performMultipleAbort, setTitle } from '../../../utils/Utils';
+
+/* tslint:disable:no-string-literal */
 interface IOwnProps {
   filterClick: (statusArray: ProcessInstanceState[] | string[]) => void;
   filters: filterType;
   setFilters: (filters) => void;
   initData: any;
   setInitData: (initData) => void;
-  abortedObj: any;
-  setAbortedObj: any;
-  setCompletedMessageObj: any;
-  setAbortedMessageObj: any;
+  selectedInstances: any;
+  setSelectedInstances: any;
   getProcessInstances: (options: any) => void;
   setSearchWord: (searchWord: string) => void;
   searchWord: string;
   isAllChecked: boolean;
   setIsAllChecked: (isAllChecked: boolean) => void;
   setSelectedNumber: (selectedNumber: number) => void;
   selectedNumber: number;
-  statusArray: string[];
+  statusArray: GraphQL.ProcessInstanceState[];
   setStatusArray: (stautsArray) => void;
-  setModalTitle: (modalTitle: string) => void;
-  setTitleType: (titleType: string) => void;
-  handleAbortModalToggle: () => void;
 }
+interface IOperationResults {
+  [key: string]: {
+    successInstances: ProcessInstanceBulkList;
+    failedInstances: ProcessInstanceBulkList;
+    ignoredInstances: ProcessInstanceBulkList;
+  };
+}
+interface IOperations {
+  [key: string]: {
+    results: {
+      successInstances: ProcessInstanceBulkList;
+      failedInstances: ProcessInstanceBulkList;
+      ignoredInstances: ProcessInstanceBulkList;
+    };
+    messages: {
+      successMessage: string;
+      warningMessage?: string;
+      ignoredMessage: string;
+      noProcessMessage: string;
+    };
+    functions: {
+      perform: () => void;
+      changeProcessStatus?: (successInstances: ProcessInstanceBulkList) => void;
+    };
+  };
+}
```

Comment:
I would suggest following changes:
- introduce singular variant of the interface too and reuse in the plural variant.
- keep IOperationResult close to the rest of interfaces + keep just the instances there.
- adjust to pass around IOperation instead of IOperationResult

```suggestion
export interface IOperationResult {
  successInstances: ProcessInstanceBulkList;
  failedInstances: ProcessInstanceBulkList;
  ignoredInstances: ProcessInstanceBulkList;
}

interface IOperationResults {
  [key: string]: IOperationResult
}

export interface IOperation {
  results: IOperationResult;
  messages: {
    successMessage: string;
    warningMessage?: string;
    ignoredMessage: string;
    noProcessMessage: string;
  };
  functions: {
    perform: () => void;
    changeProcessStatus?: (successInstances: ProcessInstanceBulkList) => void;
  };
}

interface IOperations {
  [key: string]: IOperation
}
```

Reply:
done